src/ChannelBase.php

```diff
@@ -2,20 +2,127 @@
 
 namespace Drupal\share_light;
 
+/**
+ * A base class for social media channels.
+ */
 abstract class ChannelBase implements ChannelInterface {
   protected $block;
   protected $options;
-  public function __construct($block, $options = array()) {
+
+  /**
+   * Constructor function.
+   *
+   * @param Block $block
+   *   The corresponding `share_light` `Block`.
+   * @param array $options
+   *   An array of default values.
+   */
+  public function __construct(Block $block, array $options = array()) {
     $this->block = $block;
     $this->options = $options + static::defaults();
   }
+
+  /**
+   * Returns the default values for the channel.
+   */
   public static function defaults() {
     return array('toggle' => 1);
   }
+
+  /**
+   * Adds a widget containing the options for the channel.
+   *
+   * @return array
+   *   An empty array.
+   */
   public static function optionsWidget(array &$element, array $options) {
     return array();
   }
+
+  /**
+   * Returns whether the Channel is enabled or not.
+   */
   public function enabled() {
     return !empty($this->options['toggle']);
   }
+
+  /**
+   * Fetches the node object and query parameters we want to share.
+   *
+   * In case of `campaigninon_thankyou_page`s, the parent node
+   * will be loaded (e.g. the corresponding petition).
+   *
+   * @return array
+   *   An array containing the node object and the URL query.
+   */
+  protected function getNodeAndQuery() {
```

Comment:
I think this should be refactored into two functions. `getSharedNode()` and `getQueryParameters()` instead of doing both in one. Especially if there are cases when only one of them is used.

Perhaps we could even put most of this code into `fromCurrentPath()` as it again derives a node from the current environment. With proper dependency injection this should be either a `__constructor()` argument or set using as setter-method. This would also force us to make the fact that it’s (possibly) two different nodes more visible.